image_cloudevents/modelmonitor.ipynb

```diff
@@ -0,0 +1,303 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Initializing model monitor"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mm=DkubeModelmonitor(deployemnt_id = DEPLOYMENT_ID)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mm.update_modelmonitor_basics(model_type=ModelType.Classification.value, \n",
+    "                               input_data_type=DataType.Image.value,\n",
+    "                               data_timezone=TimeZone.UTC.value)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mm.update_image_data_shape(height=200, width=200, channel=1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "DKUBE_BASE_DATASET = \"chest-xray-data\"\n",
+    "training_data = f'{DKUBE_BASE_DATASET}:'+ DKUBEUSERNAME\n",
+    "train_data_version = 'v1:'+api.get_dataset_versions(\n",
+    "    DKUBEUSERNAME,\n",
+    "    DKUBE_BASE_DATASET)[0]['version']['uuid']\n",
+    "prediction_data = LIVE_DATASET + \":\"+ DKUBEUSERNAME\n",
+    "labelled_data = LIVE_DATASET + \":\"+ DKUBEUSERNAME\n",
+    "predict_data_format = \"cloudeventlogs\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Adding train, predict, and labelled datasets"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mm.add_datasources(data_class=DatasetClass.Train.value,name=training_data,\n",
+    "                   version=train_data_version,\n",
+    "                   s3_subpath=DEPLOYMENT_ID)\n",
+    "mm.add_datasources(data_class=DatasetClass.Predict.value,name=prediction_data,\n",
+    "                   s3_subpath=DEPLOYMENT_ID, data_format=predict_data_format,\n",
+    "                   date_suffix=\"yyyy/mm/dd/hh\")\n",
+    "mm.add_datasources(data_class=DatasetClass.Labelled.value,name=labelled_data,\n",
+    "                   data_format=DataType.Tabular.value, s3_subpath=DEPLOYMENT_ID + \"/livedata\",\n",
+    "                   predict_col=\"output\",groundtruth_col=\"label\",timestamp_col=\"timestamp\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Health Monitoring"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "deployment = api.get_deployment(DEPLOYMENT_ID)\n",
+    "if ((deployment.data.imported_deployment and deployment.data.imported_deployment.cluster)\n",
+    "    or\n",
+    "    not deployment.data.imported_deployment):\n",
+    "    mm.update_deployment_monitoring_details(enabled=True, frequency=1, collect_metrics=True)"
```

Comment:
This example we know whether it is imported or not right? Can we remove all these checks?

Can we make collect_metrics True inside sdk function? We don't take it from the user

Reply:
okay, We can remove the argument from SDK and make it set it True internally.